tests/storage/ext/test_ext_sr.py

```diff
@@ -66,6 +68,32 @@ def test_snapshot(self, vm_on_ext_sr):
         finally:
             vm.shutdown(verify=True)
 
+    @pytest.mark.small_vm
+    @pytest.mark.parametrize("vdi_op", ["snapshot", "clone"])
+    def test_coalesce(self, unix_vm: 'VM', vdi_on_ext_sr: 'VDI', vdi_op):
+        vm = unix_vm
+        vdi = vdi_on_ext_sr
+        vm.connect_vdi(vdi, 'xvdb')
+        new_vdi = None
+        try:
+            vm.start()
+            vm.wait_for_vm_running_and_ssh_up()
+            install_randstream(vm)
+            vm.ssh("randstream generate -v /dev/xvdb")
+            vm.ssh("randstream validate -v --expected-checksum 65280014 /dev/xvdb")
```

Comment:
Where are you getting the expected checksum that is hardcoded?
There is a seed on the second generate call a few lines later but not here.